### DISCUSSION TITLE
Ramp demo, gauge is not moving

### DISCUSSION THREAD

**Discussion by ryzen3100** [ImpulseAdventure/GUIslice#333] [General]
Hi!
I am trying to make a simple RampGauge that goes from 0 to 99 in increments, but the gauge is not moving at all.
LCD module: 0.9" OLED (I2C) connected to NodeMCU.

The code :

> void loop()
> {
> for (int i = 0; i <= 99; i++) {
>       gslc_ElemXRadialSetVal(&m_gui, m_pElemRadial1, i);
>       Serial.println(i);
>       delay(100);
>     }
>     gslc_Update(&m_gui);
> }
> 

In the serial monitor I can see the counting works just fine.

**Comment by bcodermicro** [comment]
Are you running the loop all at once without updating the screen incementally....thus you are seeing the dial at 0 and 100 and notthing in between..move the Upddate inside the FOR with a delay....

**Reply by ryzen3100** [reply]
Wow that was a quick reply 👍 
I have moved the _gslc_Update(&m_gui);_ inside the FOR but the gauge is still stuck

**Reply by ryzen3100** [reply]
An update, not sure what am I doing wrong, the same code works with the progress bar but not with the ramp gauge.
Code:
```
void loop()
{
  for (int i = 0; i <= 99; i++) {
    gslc_ElemXProgressSetVal(&m_gui, m_pElemProgress1, i);
    Serial.println(i);
    delay(100);
    gslc_Update(&m_gui);
  }
}

```

**Reply by ImpulseAdventure** [reply]
Hi @ryzen3100 -- just to confirm, I think you are saying that ElemXProgressSetVal() + Update() works... but was it ElemXRadialSetVal() or ElemXRampSetVal() that didn't appear to be working for you?

**Reply by ryzen3100** [reply]
Hi @ImpulseAdventure, Ohh yes I meant ElemXRadialSetVal() was not working, sorry got a little confused.
I have re-downloaded the library, manually added fork [WIP248-SSD1306_Ada_I2C](https://github.com/ImpulseAdventure/GUIslice/compare/WIP248-SSD1306_Ada_I2C) and ElemXRadialSetVal() is working great now.

PS.: Does this project have a forum website or a discord server?

**Reply by ImpulseAdventure** [reply]
Hey @ryzen3100 -- Note that the WIP248 fork was an earlier attempt at adding support for the SSD1306 I2C mode.
I have recently integrated I2C mode support into the main branch (and also added support for new resolution modes tonight). Please try out the latest GUIslice and use the new `ard-adagfx-ssd1306-notouch` config, and make the following mods to select I2C mode:
- comment out the DRV_DISP_ADAGFX_SSD1306_SPI line
- uncomment the DRV_DISP_ADAGFX_SSD1306_I2C* lines

Please let me know if that works and whether you still observe the SetVal issue.
thx
